BigQuery service account docs (#12710)
metabase/metabase: docs/administration-guide/databases/bigquery.md

## Working with Google BigQuery in Metabase

This page provides information on how to create and manage a connection to a Google [Bigquery](https://cloud.google.com/bigquery) dataset.

> **Starting in v0.15.0**, Metabase provides a driver for connecting to BigQuery directly and executing queries against any datasets you have.  

> >**Starting in v0.36.0**, Metabase allows you to connect to your BigQuery dataset using a [Service Account](https://cloud.google.com/iam/docs/understanding-service-accounts).

Reviewer: Because we version docs, we typically haven't included references to specific versions like this.

Author: The 0.15 notice was in the current docs, so I followed convention. But I agree: no need to disclose that info here.

2. **Fill out the service account details**. Name the service account, and add a description (the service account ID will populate once you add a name.) Then click the **Create** button.

3. **Grant service account access to this project**. You'll need to add **roles** to the service account so that Metabase will have permission to view and run queries against your dataset. Make sure you add the following roles to the service account:

    - BigQuery Data Viewer
    - BigQuery Metadata Viewer
    - BigQuery Job User (distinct from BigQuery User)

    For more information on **roles** in BigQuery, see [Google Cloud Platorm's documentation](https://cloud.google.com/bigquery/docs/access-control).

4. **Create key**. Once you have assigned roles to the service account, click on the **Create Key** button, and select **JSON** for the **key type**. The JSON file will download to your computer.

> **You can only download the key once**. If you delete the key, you'll need to create another service account with the same roles.

## Metabase: adding a BigQuery dataset

> Once you have your created and downloaded your service account JSON file for your BigQuery dataset, head over to your Metabase instance, click on the **settings** cog, and select **Admin** to bring up Admin mode. In the **Databases** section, click on the **Add database** button in the upper right. (You can also access this page from the **Settings** menu, under **Setup** and **GET CONNECTED**.)

Reviewer: > You can also access this page from the **Settings** menu, under **Setup** and **GET CONNECTED**.)

I don't think you need this line. They don't really need to know this.

On the **ADD DATABASE** page, select **BigQuery** from the **Database type** dropdown. Metabase will present you with the relevant configuration fields to fill out:

![images](../images/bigquery_add_database.png)

> ### Fields

Reviewer: This separation of "fields" vs "sliders" seems odd and unnecessary to me. I think you can just list all the settings in order without breaking them out like this.

Author: Agreed. I'll put everything under an umbrella **Settings** section.

#### Name

**Name** is the title of your database in Metabase.

#### Dataset ID

Each BigQuery dataset will have a **Dataset ID**. You can find this ID via the [Google Cloud Console](https://console.cloud.google.com/). If you're not sure where to find the Dataset ID, see Google's documentation on [getting information on datasets](https://cloud.google.com/bigquery/docs/dataset-metadata#getting_dataset_information).

> When entering the Dataset ID, omit the Project ID prefix. For example, if your ID is `project_name:dataset_id`, only enter `dataset_id`.

#### Service account JSON file

Upload the service account JSON file you created when following the [steps above](#google-cloud-platform:-creating-a-service-account-and-json-file). The JSON file contains the credentials your Metabase application will need to read and query your dataset, as defined by the **roles** you added to the service account. If you need to add additional roles, you have to create another service account, download the JSON file, and upload the file to Metabase.

### Sliders

#### Use the Java Virtual Machine (JVM) timezone

> _Default: Disabled_

Reviewer: What's the rationale behind documenting the default state, particularly since we give advice below about leaving it off?

Author: This text is provided in the UI itself. I'm including it in the docs as well so that readers will have the information even when they are not following along while configuring a connection (for example, if they are exploring our docs to gauge our support for BigQuery).

We suggest you leave this off unless you're doing manual timezone casting in many or most of your queries with this data.

#### Automatically run queries when doing simple filtering and summarizing.

_Default: Enabled_

When this slider is on, Metabase will automatically run queries when users do simple explorations with the Summarize and Filter buttons when viewing a table or chart. You can turn this off if querying this database is slow. This setting doesn’t affect drill-throughs or SQL queries.

#### This is a large database, so let me choose when Metabase syncs and scans

_Default: Disabled_

By default, Metabase does a lightweight hourly sync and an intensive daily scan of field values. If you have a large database, we recommend turning this on and reviewing when and how often the field value scans happen.

### Save your database configuration

When you're done, click the **Save** button. A modal should pop up, informing you that your database have been added.

![Database added](../images/database-added.png)

> You can click on **Explore this data** to have **Metabot** summarize the tables in your dataset, or click on the **I'm good thanks** to return to **Metabase Admin** mode.

Reviewer: > to have **Metabot** summarize the tables in your dataset

Suggest replacing with "to see some automatic explorations of your data"

## Working with Google BigQuery in Metabase

> This page provides information on how to create and manage a connection to a Google [Bigquery](https://cloud.google.com/bigquery) dataset.

Reviewer: ```suggestion
This page provides information on how to create and manage a connection to a Google [BigQuery](https://cloud.google.com/bigquery) dataset.
```

#### Use the Java Virtual Machine (JVM) timezone

_Default: Disabled_

We suggest you leave this off unless you're doing manual timezone casting in many or most of your queries with this data.

#### Automatically run queries when doing simple filtering and summarizing.

_Default: Enabled_

When this slider is on, Metabase will automatically run queries when users do simple explorations with the Summarize and Filter buttons when viewing a table or chart. You can turn this off if querying this database is slow. This setting doesn’t affect drill-throughs or SQL queries.

#### This is a large database, so let me choose when Metabase syncs and scans

_Default: Disabled_

By default, Metabase does a lightweight hourly sync and an intensive daily scan of field values. If you have a large database, we recommend turning this on and reviewing when and how often the field value scans happen.

### Save your database configuration

> When you're done, click the **Save** button. A modal should pop up, informing you that your database have been added.

Reviewer: ```suggestion
When you're done, click the **Save** button. A modal should pop up, informing you that your database has been added.
```

![Database added](../images/database-added.png)

> You can click on **Explore this data** to see some automatic explorations of your data, or click on the **I'm good thanks** to return to **Metabase Admin** mode.

Reviewer: ```suggestion
You can click on **Explore this data** to see some automatic explorations of your data, or click **I'm good thanks** to stay in the Admin Panel.
```

SELECT *
FROM [my_dataset.my_table]
```

> ## Troubleshooting

Reviewer: We could also recommend they check out this troubleshooting guide for connecting to databases and data warehouses: https://www.metabase.com/docs/latest/troubleshooting-guide/datawarehouse.html

SELECT *
FROM [my_dataset.my_table]
```

## Troubleshooting

If you're having trouble with your BigQuery connection, check out [Metabase's discussion forum](https://discourse.metabase.com/search?q=bigquery) to see if someone has encountered and resolved a similar issue.

## Further reading

> - [Managing databases](https://www.metabase.com/docs/latest/administration-guide/01-managing-databases.html).

Reviewer: This link is a bit redundant with the current article. It covers a lot of the same ground.

- [Metadata editing](https://www.metabase.com/docs/latest/administration-guide/03-metadata-editing.html).
- [Creating segments and metrics](https://www.metabase.com/docs/latest/administration-guide/07-segments-and-metrics.html).
- [Setting data access permissions](https://www.metabase.com/docs/latest/administration-guide/05-setting-permissions.html).

> - [Creating collections](https://www.metabase.com/docs/latest/administration-guide/06-collections.html).

Reviewer: Not sure this collection article feels like a natural related article here.
